Libs/base/rand.slang

```diff
@@ -0,0 +1,45 @@
+pub class Rand:
+	"""
+	32 bit PCG random number generator
+
+	Really minimal PCG32 code, ported to slang lang.
+	Original C code by (c) 2014 M.E. O'Neill / pcg-random.org
+	Licensed under Apache License 2.0 (NO WARRANTY, etc. see website)
+	"""
+	pub var state: uint64 = uint64(0)
+	pub var inc: uint64 = uint64(0)
+
+	pub fn next() -> uint32:
+		"""
+		Generate a random 32 bit number
+		range: 0 <= x < 2**32
+		"""
+		let old_state = state
+		state = uint64(6364136223846793005) * old_state + (inc | uint64(1))
```

Comment:
This large number gives issues in the current implementation of 64 bits integers in combination with boxing/unboxing. See `slangrt_box_int64`.

A few options to fix this:
- malloc 8 bytes for 64 bits integers
- Disallow 64 bits integer to be boxed at all
- Use smaller integer number for random generator